sandbox/cgroup/monitor.go

```diff
@@ -0,0 +1,92 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup
+
+import (
+	"path"
+
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/sandbox/cgroup/inotify"
+	"github.com/snapcore/snapd/strutil"
+)
+
+// MonitorFullDelete allows to monitor a group of files/folders
+// and, when all of them have been deleted, emits the specified name through the channel.
+func MonitorFullDelete(name string, folders []string, channel chan string) error {
+	wd, err := inotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+
+	var toMonitor []string
+	var tmpFolders []string
+
+	for _, fullPath := range folders {
+		basePath := path.Dir(fullPath)
+		if !strutil.ListContains(toMonitor, basePath) {
+			err := wd.AddWatch(basePath, inotify.InDelete)
```

Comment:
IN_DELETE will fire whenever children of the watched path are deleted. If you only care about when the cgroup directory itself is deleted, then perhaps IN_DELETE_SELF is appropriate?

Reply:
Good point! And it simplifies the code.